pkg/controller/manilacsi/manila_credentials.go

```diff
@@ -0,0 +1,134 @@
+package manilacsi
+
+import (
+	"context"
+	"fmt"
+
+	manilacsiv1alpha1 "github.com/Fedosin/csi-driver-manila-operator/pkg/apis/manilacsi/v1alpha1"
+	"github.com/go-logr/logr"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	//"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	cloudsSecretKey     = "clouds.yaml"
+	installerSecretName = "installer-cloud-credentials"
+	driverSecretName    = "csi-manila-secrets"
+	secretNamespace     = "manila-csi"
+	cloudName           = "openstack"
+)
+
+func (r *ReconcileManilaCSI) handleManilaCrednetials(instance *manilacsiv1alpha1.ManilaCSI, reqLogger logr.Logger) error {
+	reqLogger.Info("Reconciling Manila Credentials")
+
+	found := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: driverSecretName, Namespace: secretNamespace}, found)
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
Can you turn this into an early return, to avoid big nested blocks and `else`s? Like (feel free to improve! just an example):

```Go
if err == nil {
    reqLogger.Info("Skip reconcile: Secret already exists", "Secret.Namespace", found.Namespace, "Service.Name", found.Name)
    return nil
}

if err != nil && !errors.IsNotFound(err) {
    return err
}

// From here, errors.IsNotFound(err) == true
cloudConfig, err := r.getCloudFromSecret()
[...]
```

Reply:
yeah, why not :)